crates/wasmtime/src/func.rs

```diff
@@ -191,6 +203,12 @@ pub(crate) struct FuncData {
     ty: Option<Box<FuncType>>,
 }
 
+// Safety: the `in_store_func_ref` field always points to something inside this
+// `FuncData`'s store, which is only ever accessed when we know that we are on
+// the same thread as the store.
+unsafe impl Send for FuncData {}
+unsafe impl Sync for FuncData {}
```

Comment:
I worry about this because it can accidentally paper over non-send/sync in `kind: FuncKind` and `ty: Option<Box<FuncType>>`. The latter is probably tested elsewhere but there's not a ton of protection here against `FuncKind`.

I don't have a great way to solve this other than unergonomic wrappers, but I'd probably still prefer that to the impl here on the whole struct.